scripts/config/generate-config.js

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Configuration generation wrapper script
+ * This script:
+ * 1. Runs the generate_config.sh bash script to clone the schema repository
+ * 2. Uses json-schema-to-typescript to generate TypeScript types from the JSON schema
+ * 3. Post-processes the generated file by adding license headers and fixing types
+ */
+
+const { compileFromFile } = require('json-schema-to-typescript');
+const fs = require('fs');
+const { execSync } = require('child_process');
+const path = require('path');
+
+const SCRIPT_DIR = __dirname;
+const ROOT_DIR = path.join(SCRIPT_DIR, '..', '..');
+
+// Run the bash script to get the latest version of the schema
+const scriptPath = path.join(SCRIPT_DIR, 'generate-config.sh');
```

Comment:
Since you are setting the `cwd` of the child process, it might be redundant to compute the absolute path of the script like this.